### PR TITLE
Gnutls 3.4.x

### PR DESCRIPTION
**Pull request by rakkesh**
https://github.com/Homebrew/homebrew/issues/46562

### CODE REVIEW

File: Library/Formula/wireshark.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ class Wireshark < Formula
     url "https://www.wireshark.org/download/src/all-versions/wireshark-1.12.8.tar.bz2"
     mirror "https://1.eu.dl.wireshark.org/src/wireshark-1.12.8.tar.bz2"
     sha256 "357e0a4e49525d80cdc740bb16539fcdb526ad38cc2ed6cabedafc9bdee5c7e7"
+    revision 1
```
Comment (bfontaine) [inline Library/Formula/wireshark.rb]: Please put the revision at the top-level, just after `homepage`.

Comment (mikemcquaid) [inline Library/Formula/wireshark.rb]: Let's fix this up when we pull instead to avoid having another (long) rebuild.

Comment (DomT4) [inline Library/Formula/wireshark.rb]: Build fails without the line in the right place now.

Comment (DomT4) [inline Library/Formula/wireshark.rb]: > ==> brew readall --aliases --syntax
==> FAILED
Error: problem in /usr/local/Library/Formula/wireshark.rb
undefined method `revision' for #<SoftwareSpec:0x007fe7cd534730>

Comment (mikemcquaid) [inline Library/Formula/wireshark.rb]: Sorry, I misread the diff (didn't click expand). Doh.

Comment (rakkesh) [inline Library/Formula/wireshark.rb]: Let me know, if I need to make the changes now or you guys are handling it now.

Comment (DomT4) [inline Library/Formula/wireshark.rb]: Aye, you'll need to follow Baptiste's request here :). `revision` lines are always in the initial block, essentially.

**Comment by DomT4** [comment]
Aborted the build because Nettle failed, and then because so much else uses it it was trying and failing to build Nettle every time, and that would have continued for hours heh. Mind checking out the `nettle` failure? As far as I know the one in Versions passed at the time, so not sure why failing now.

**Comment by DomT4** [comment]
Link: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/38355/version=mavericks/console

**Comment by rakkesh** [comment]
I had interchanged the order of `make install` and `make check`, hence the error.

```
dyld: Library not loaded: /usr/local/Cellar/nettle/3.1/lib/libhogweed.4.dylib
  Referenced from: /private/tmp/nettle20151216-35061-14s3pn4/nettle-3.1/testsuite/./aes-test
  Reason: image not found
../run-tests: line 57: 38140 Trace/BPT trap: 5       "$1" $testflags
```

I have now reverted the change.

**Comment by DomT4** [comment]
Ah, Thanks!

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Looks like `p11-kit` needs to be explicitly disabled or updated in this PR. There are a few other failures that need chasing up on as well.

**Comment by rakkesh** [comment]
`p11-kit` is not enabled by default, https://github.com/rakkesh/homebrew/blob/0c5d3290574893a2999de30a4e2661f952bae6f3/Library/Formula/gnutls.rb#L49-L51. 

`gnu-smalltalk` audit warnings seem to be introduced in a recent commit. I'll change the formula accordingly.

As for the rest of the failures, there are local errors on the build machine.
```
==> Verifying task-2.5.0.tar.gz checksum
tar xf /Library/Caches/Homebrew/task-2.5.0.tar.gz
Error: No such file or directory - /private/tmp/task20151216-16350-bekp5t/task-2.5.0/src/tw
```

`ncdc` error is weird as there are no signs of a failure but the report says otherwise,
http://bot.brew.sh/job/Homebrew%20Pull%20Requests/38362/version=mavericks/testReport/junit/brew-test-bot/mavericks/install_ncdc/

**Comment by rakkesh** [comment]
@DomT4 updated `gnu-smalltalk` formula with `--with-lispdir=#{share}/emacs/site-lisp/#{name}`. I hope the build machine errors are handled by the admin.

**Comment by apjanke** [comment]
The build errors this time around are timeouts, from having so many formulae involved. The stuff it actually got through all succeeded.

**Comment by DomT4** [comment]
Not quite: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/38394/version=el_capitan/consoleFull. Errors aren't shown, but there are some there.

**Comment by DomT4** [comment]
`p11-kit` again, `ncdc` failed, `ntopng` failed, `task` failed. The last two failures may not be specific to this PR, they look a bit odd.

**Comment by apjanke** [comment]
Ah, I missed that... looks like the timeout message at the end prevents accumulated failures from being reported down there, so you have to scan through? Makes sense; the task being aborted is the one tracking the failures. Sorry about that.

I'm not sure if I'm reading the logs right, but it seems like `ncdc` is failing because `gnutls` itself is failing because it didn't find a version of `p11-kit` it likes?

```
checking for P11_KIT... no
configure: error: 
***
*** p11-kit >= 0.23.1 was not found. To disable PKCS #11 support
*** use --without-p11-kit, otherwise you may get p11-kit from
*** http://p11-glue.freedesktop.org/p11-kit.html
*** 
```

Homebrew's `p11-kit` formula is at 0.18.4.

The logs say it is trying to use `p11-kit` here; you can see the `--with-p11-kit` in the command line. This particular failure happens when it's installing `gnutls` as a dependency for `ncdc`, not in the main build, I think.

```
==> Installing ncdc dependency: gnutls
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20151216-68989-1xrsgs4.sb nice /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Library/Homebrew -- /usr/local/Library/Homebrew/build.rb /usr/local/Library/Formula/gnutls.rb --verbose --with-p11-kit
==> Downloading ftp://ftp.gnutls.org/gcrypt/gnutls/v3.4/gnutls-3.4.7.tar.xz
Already downloaded: /Library/Caches/Homebrew/gnutls-3.4.7.tar.xz
==> Verifying gnutls-3.4.7.tar.xz checksum
==> ./configure --disable-dependency-tracking --disable-silent-rules --disable-static --prefix=/usr/local/Cellar/gnutls/3.4.7 --sysconfdir=/usr/local/etc --with-default-trust-store-file=/usr/local/etc/openssl/cert.pem --disable-heartbeat-support
checking build system type... x86_64-apple-darwin13.4.0
checking host system type... x86_64-apple-darwin13.4.0
```

Looks like that's because `ncdc` [has `depends_on "gnutls" => "with-p11-kit"`](https://github.com/Homebrew/homebrew/blob/f16987d7090a8c5b9ad1fc8d6a93c8b2cac6a440/Library/Formula/ncdc.rb#L26).

Maybe we need to version-bump `p11-kit` as part of this PR?

**Comment by apjanke** [comment]
And maybe the commits in this PR should be squashed to a single commit, since they are interdependent on each other's changes? That way each commit in the history would still have a functioning set of formulae.

**Comment by DomT4** [comment]
I thought the CI showed failures on timeout regardless, but perhaps at least one of the builds has to succeed/fail in the time limit for that to happen.

**Comment by apjanke** [comment]
Cool.

The CI looks like it is still running on a 45-minute-or-so timeout, so it timed out much earlier in the build process on this, before any errors occurred.

**Comment by apjanke** [comment]
Here's an issue for us. This GNUTLS wants `p11-kit` 0.23.1 or later. But 0.23.1 and 0.23.2 are development releases, [according to the p11-kit mailing list](http://lists.freedesktop.org/archives/p11-glue/2015-February/000554.html). 

> This is an development release.
>
> What's new in 0.23.1
> --------------------
>
> * Use new PKCS#11 URI draft fields for URIs [#86474 #87582]

The p11-kit [0.22.x series is stable](http://lists.freedesktop.org/archives/p11-glue/2014-October/000504.html).

**Comment by rakkesh** [comment]
I have submitted a [PR](https://github.com/Homebrew/homebrew/pull/47951) for `ncdc`. Once, it's merged I'll rebase the current PR on top of updated master.

**Comment by apjanke** [comment]
That `ncdc` PR is merged. You can give this a rebase.

**Comment by rakkesh** [comment]
Is there anything more I need to do for moving this PR forward ?

**Comment by DomT4** [comment]
I'll push it through CI once the queue has calmed down a bit. Have self-assigned this PR as a reminder to do that later.

**Comment by rakkesh** [comment]
@DomT4 🖖. I hope this gets through this time without any hiccups. 🤘

**Comment by dunn** [comment]
I added a commit to https://github.com/Homebrew/homebrew/pull/47990 removing the p11-kit option, so this will need to be rebased after that's merged.

File: Library/Formula/gnutls.rb [hunk 2]
```diff
@@ -45,6 +45,10 @@ def install
       args << "--with-guile-site-dir=no"
     end
 
+    if build.without? "p11-kit"
```
Comment (dunn) [inline Library/Formula/gnutls.rb]: Oh nevermind, it doesn't need rebasing but this addition can be dropped.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Merged! Thank you for the persistence on this one, and another contribution to Homebrew @rakkesh! :smiley_cat:

Tiny future note is that we prefer one commit per formula changed, including on these mass revision bump situations.

**Comment by rakkesh** [comment]
Cheers 🍻

**Pull request closed without merging**